test/test_timeout.rb

```diff
@@ -172,4 +172,24 @@ def test_threadgroup
     end;
   end
 
+  def test_handling_enclosed_threadgroup
+    # The problem "add: can't move from the enclosed thread group" #24,
+    # happens when the timeout_thread is created in an enclosed ThreadGroup.
+    # Make sure there isn't a timeout_thread in use to test this situation.
+    Timeout.instance_eval { @timeout_thread = nil }
```

Comment:
Use `assert_separately` just like the test above to avoid that issue

Reply:
Just changed it to use assert_separately